sympy/parsing/sympy_parser.py

```diff
@@ -906,7 +907,49 @@ def eval_expr(code, local_dict: DICT, global_dict: DICT):
         code, global_dict, local_dict)  # take local objects in preference
     return expr
 
+def override_signature(new_sig: str):
+    """
+    Decorator to override the displayed function signature in interactive environments
+    like IPython or Jupyter notebooks.
+
+    Parameters
+    ----------
+    new_sig : str
+        A string representation of the desired function signature. It should be valid
+        as the arguments of a lambda function. Example: "x, y=1, z='foo'".
+
+    Returns
+    -------
+    decorator : Callable
+        A decorator that sets the `__signature__` attribute of the function it wraps.
+        This changes how the function appears in tools like IPython's `?` or `help()`,
+        without affecting the actual behavior or callable interface of the function.
+
+     Examples
+    --------
+    >>> @override_signature("x, y=1, z='foo'")
+    ... def my_func(x, y=1, z='foo'):
+    ...     return x + y
+
+    >>> import inspect
+    >>> str(inspect.signature(my_func))
+    '(x, y=1, z=\'foo\')'
+
+    References
+    ==========
+
+    .. [1] https://stackoverflow.com/questions/69845887/simplifying-the-calling-signature-for-repeatedly-used-arguments
+           (Simplifying the calling signature for repeatedly-used arguments)
+
+    .. [2] https://stackoverflow.com/questions/2677185/how-can-i-read-a-functions-signature-including-default-argument-values
+           (How can I read a function’s signature including default argument values?)
+    """
+    def decorator(func):
+        func.__signature__ = inspect.signature(eval(f"lambda {new_sig}: None"))
+        return func
+    return decorator
 
+@override_signature("s, local_dict=None, transformations='standard', global_dict=None, evaluate=True")
 def parse_expr(s: str, local_dict: DICT | None = None,
```

Comment:
I don't think this is a good solution for this problem. We don't want to annotate every function with a duplicate signature.